open-craft/ansible-playbooks
# [BB-4309] Reformat mongodb devices using XFS

## Description

This pull requests changes the MongoDB provisioning to format the device used for `/var/lib/mongodb`  using XFS for better WiredTiger performance.

Related PR: https://github.com/open-craft/ansible-secrets/pull/322

## Testing

1. Create a VM based on Ubuntu 20.04 or use 54.37.151.33
1. SSH into the VM and create an empy file in `/dev/md3`
    `$ sudo dd if=/dev/zero of=/dev/md3 bs=1M count=100`
1. Modify your local `/etc/hosts`  to point mongodb-secondary-1 to the VM
    `54.37.151.33    mongodb-secondary-1.net.opencraft.hosting`
1. Comment the `common-server` role in `playbooks/mongodb.yml`
1. Run the deployment
    `$ ansible-playbook deploy/playbooks/mongodb.yml -e @group_vars/mongodb/public.yml -e @host_vars/mongodb-secondary-1.net.opencraft.hosting.yml -e "MONGODB_EXPORTER_ENABLED=false"  -l mongodb-secondary-1.net.opencraft.hosting`
1. SSH into the VM and confirm `/dev/md3` has been mounted using a XFS filesystem
1. Run the playbook again to confirm it still works, just bypassing the filesystem format.
1. Now, format `/dev/md3` using EXT4
     `$ mkfs.ext4 /dev/md3`
1. Run the playbook again and see it failing.
